src/main/java/service/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,18 @@
+package service.operation;
+
+import data.db.Storage;
+import java.util.Map;
+import model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+
+    @Override
+    public void operation(FruitTransaction.Operation operation, FruitTransaction fruitTransaction) {
+        for (Map.Entry<String, Integer> fruit : Storage.fruitsStorage.entrySet()) {
+            if (fruit.getKey().equals(fruitTransaction.getFruit())) {
+                fruit.setValue(Math.max(fruit.getValue() - fruitTransaction.getQuantity(), 0));
```

Comment:
The current implementation allows the quantity of fruits to go negative, as it sets the value to 0 if the result is negative. Instead, you should check if there's enough fruit in the storage before proceeding with the transaction and throw an exception if there isn't enough.